src/reducers/columnStateHelper.js

```diff
@@ -71,9 +71,10 @@ function scrollTo(state, props, oldScrollToColumn, scrollX) {
   } = columnWidths(state);
   const fixedColumnsCount = fixedColumns.length;
 
-  const scrollToUnchanged = scrollToColumn === oldScrollToColumn
+  const noScrollableColumns = scrollableColumns.length === 0;
+  const scrollToUnchanged = scrollToColumn === oldScrollToColumn;
   const selectedColumnFixed = scrollToColumn < fixedColumnsCount;
```

Comment:
Looking again, pretty sure than an additional check needs to be here, because
`selectedColumnFixed` represents that scrollToColumn is on a fixed (to left) column. We should do the same fixedRight columns.

maybe something like:
```
const fixedColumnsCount = fixedColumns.length;
const scrollableColumnsCount = scrollableColumns.length;

const selectedColumnFixedRight = scrollToColumn >= (fixedColumnsCount + scrollableColumnsCount);
```

Reply:
Yep, without the added check it will change `srcollX `even for a `scrollTo` applied to `fixedRight` column.
there was also an issue with the calculation of `availableScrollWidth`.
Both should be fixed with the newly added commit 7da157a.